concent_api/core/payments/backends/sci_backend.py

```diff
@@ -32,31 +33,54 @@ def get_list_of_payments(
     """
     assert isinstance(requestor_eth_address,    str) and len(requestor_eth_address) == ETHEREUM_ADDRESS_LENGTH
     assert isinstance(provider_eth_address,     str) and len(provider_eth_address)  == ETHEREUM_ADDRESS_LENGTH
-    assert isinstance(payment_ts,               int) and payment_ts     >= 0
+    assert isinstance(min_block_timestamp, int) and min_block_timestamp >= 0
     assert isinstance(transaction_type,         Enum) and transaction_type in TransactionType
 
-    payment_interface: SmartContractsInterface = PaymentInterface()
+    payment_interface: SCIImplementation = PaymentInterface()
 
-    last_block_before_payment = BlocksHelper(payment_interface).get_first_block_after(payment_ts).number
+    first_block_after_payment_number = get_first_block_from_timestamp(payment_interface, min_block_timestamp).number
+    latest_block_number = payment_interface.get_block_number()
+    if latest_block_number - first_block_after_payment_number < payment_interface.REQUIRED_CONFS:
+        return []
```

Comment:
I'm wondering if it won't be better that if this would be a variable?

Reply:
Will this ever change? What could actually cause this situation? Maybe exception is better then? And maybe logging?

Reply:
https://github.com/golemfactory/concent/issues/960#issuecomment-453585799